src/tigerbeetle/main.zig

```diff
@@ -260,13 +259,12 @@ const Command = struct {
         } });
         defer message_pool.deinit(allocator);
 
-        var aof: AOFType = undefined;
-        if (constants.aof_record) {
+        var aof: ?AOF = if (args.aof) blk: {
             const aof_path = try std.fmt.allocPrint(allocator, "{s}.aof", .{args.path});
             defer allocator.free(aof_path);
 
-            aof = try AOF.from_absolute_path(aof_path);
-        }
+            break :blk try AOF.from_absolute_path(aof_path);
+        } else null;
 
```

Comment:
```suggestion
        } else null;
        defer if (aof != null) aof.?.deinit();
```

Reply:
Good catch!

Reply:
Actually maybe not :laughing: .

The AOF doesn't have a `deinit()`. It does have `close()` which closes the file.

Reply:
Close actually make more sense!